### PR TITLE
[IMP] mail: raise error when manually fetch mail

### PR DESCRIPTION
**Pull request by daiduongnguyen-odoo**
-Problem: when fetching mail using cron , totally agree that we must not raise error, but only log info is not enough for user to know which kind of error they get when fetching mail (error can be connection problem, wrong configuration at server,...)
-Solution: raise the error when user hit the button fetch mail manually in order to have a clear view of errors

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by duong77476-viindoo** [comment]
Hi @std-odoo Can you take a look please?

**Comment by duong77476-viindoo** [comment]
Hi @pyduf sorry to bother you but can you take a look whether this PR is necessary or not ?

**Review by reth-odoo — COMMENTED** [review]
Hi

Actually I'm wondering why logging with exc_info isn't enough. Do we want to catch the error at some point? Or is exc_info lacking some context?

**Comment by duong77476-viindoo** [comment]
> Hi
> 
> Actually I'm wondering why logging with exc_info isn't enough. Do we want to catch the error at some point? Or is exc_info lacking some context?

Because i have to search the all log instead i can view directly in odoo.
Problem that cause me do this PR is my client recently fetch mail normally but then one day they see that "Why mail won't come anymore" and then they try to fetch mail manually and nothing happen because we only have logging info, they contact me and i have to download the logs and search the error which will cause sometime. Turnout error is from their mail server

File: addons/mail/models/fetchmail.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def button_confirm_login(self):
     @api.model
     def _fetch_mails(self):
         """ Method called by cron to fetch mails from servers """
-        return self.search([('state', '=', 'done'), ('server_type', '!=', 'local')]).fetch_mail()
+        return self.search([('state', '=', 'done'), ('server_type', '!=', 'local')]).with_context(call_by_cron=True).fetch_mail()
```
Comment (std-odoo) [inline addons/mail/models/fetchmail.py:185]: it's not a big deal imo, can we do it in master instead with a new argument to `fetch_mail` which will be `raise_exception` set at `False` by default ?

A bit like the method `_send` on `mail.mail`

What do you think ?

Also the commit message can be improved a bit (we generally try to not exceed 72 chars by line, do not give your opinion like "I agree" by only give facts, etc)

**Comment by std-odoo** [comment]
also not sure we should just raise the exception like this, it might give a traceback in the interface

File: addons/mail/models/fetchmail.py (hunk 1 above)
Comment (duong77476-viindoo) [reply addons/mail/models/fetchmail.py:185]: Ok , i get it, let close this PR and move to master later. Thank for your response

**Comment by duong77476-viindoo** [comment]
> also not sure we should just raise the exception like this, it might give a traceback in the interface

Yes , you 're right , need to create a friendly exception

**Comment by duong77476-viindoo** [comment]
replace by https://github.com/odoo/odoo/pull/152637 cc @std-odoo

**Pull request closed without merging**